unstoppable-ios-app/domains-manager-ios/Modules/PurchaseDomains/Checkout/PurchaseDomainsCheckoutView.swift

```diff
@@ -307,10 +318,43 @@ private extension PurchaseDomainsCheckoutView {
 private extension PurchaseDomainsCheckoutView {
     func onAppear() {
         checkoutData = purchaseDomainsPreferencesStorage.checkoutData
+        didSelectWallet(selectedWallet)
     }
     
     func didSelectWallet(_ wallet: WalletWithInfo) {
-        
+        Task {
+            selectedWallet = wallet
+            isLoading = true
+            do {
+                try await purchaseDomainsService.authoriseWithWallet(wallet.wallet,
+                                                                     toPurchaseDomains: [domain])
+            } catch {
+                Debugger.printFailure("Did fail to authorise wallet \(wallet.wallet.address) with error \(error)")
+            }
+            isLoading = false
+        }
+    }
+    
+    func purchaseDomains() {
```

Comment:
Just an idea:

a method that starts async process to be named: startPurchaseDomains? purchaseDomainsAsync?